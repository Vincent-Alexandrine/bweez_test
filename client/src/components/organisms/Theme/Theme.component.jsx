import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';

import ThemeService from 'Src/theme';
import vManuscript from 'Styles/vManuscript';

import GlobalStyle from './Theme.styled.js';

class Theme extends Component {
  static defaultProps = {
    currentTheme: vManuscript.name,
  };

  static propTypes = {
    currentTheme: PropTypes.string,
    children: PropTypes.node,
  };

  constructor(props) {
    super(props);
    ThemeService.add(vManuscript);
  }

  render() {
    const { children, currentTheme } = this.props;
    const theme = ThemeService.get(currentTheme);
    return (
      <ThemeProvider theme={theme}>
        <Fragment>
          <GlobalStyle />
          {children}
        </Fragment>
      </ThemeProvider>
    );
  }
}

export default Theme;
