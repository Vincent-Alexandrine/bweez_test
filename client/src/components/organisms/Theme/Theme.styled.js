import { createGlobalStyle } from 'styled-components';

import globalTheme from 'Styles/vManuscript/global';

export default createGlobalStyle`
  ${globalTheme}
`;
