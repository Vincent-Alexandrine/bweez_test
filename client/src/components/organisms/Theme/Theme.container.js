import ReduxWrapper from 'Src/utilities/methods/redux.router.wrapper.js';
import Theme from './Theme.component.jsx';

// define Component's props from Store
// const mapStateToProps = (state) => ({
//  current: DefineCurrent(state.status, state.smt),   <-- pass props "current" to Component
// });

const mapStateToProps = (state) => ({
  currentTheme: state.theme.current,
});

// define Component's cb as props, triggering "dispatch"
// const mapDispatchToProps = (dispatch) => ({
//   onSelect: id => dispatch(selectSmt(id)),   <-- pass props "onSelect" to Component
// });

const mapDispatchToProps = () => ({

});

export default ReduxWrapper(
  mapStateToProps,
  mapDispatchToProps
)(Theme);
