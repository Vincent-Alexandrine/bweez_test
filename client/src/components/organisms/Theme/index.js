import Theme from './Theme.container.js';

export default Theme;
