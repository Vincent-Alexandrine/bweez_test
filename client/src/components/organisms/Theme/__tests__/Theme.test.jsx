import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Theme from '../Theme.component.jsx';
import Styled from '../Theme.styled.js';

test('Theme component should match snapshot', () => {
  const component = renderer
    .create(<Theme />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('Theme styled component should match snapshot', () => {
  const component = renderer
    .create(<Styled />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
