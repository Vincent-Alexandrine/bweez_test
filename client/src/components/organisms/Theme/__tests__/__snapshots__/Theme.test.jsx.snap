// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme component should match snapshot 1`] = `null`;

exports[`Theme styled component should match snapshot 1`] = `null`;
